tcmur_cmd_handler.c

```diff
@@ -745,6 +745,18 @@ static int tcmur_cmd_handler(struct tcmu_device *dev, struct tcmulib_cmd *cmd)
 
 	track_aio_request_start(rdev);
 
+	/*
+	 * Try to passthrough the cmds to the device.
+	 */
+	if (rhandler->handle_cmd) {
+		ret = handle_passthrough(dev, cmd);
+		if (ret != TCMU_NOT_HANDLED)
```

Comment:
Yes it will be a pain.

Exposing an array of commands is a good idea, but IMO to simplify this, if the handler is not passthrough only type && it has a handle_cmd callout, it should mean that the handler want to handle some commands by itself, and we should try handle_cmd callout first.

And all the handler_cmd callout should obey a rule that only when the cmds are not supported by it should return TCMU_NOT_HANDLED. Then the runner will try to handle the cmds in generic.

Thanks,